scripts/dss_v2_apdapter.py

```diff
@@ -0,0 +1,466 @@
+                    try:
+                        data_file_old_name = file_contents['file_core']['file_name']
+                    except KeyError as e:
+                        raise Exception(bundle_fqid, manifest_entry['name'],
+                                        f"'file_core.file_name' not found in file") from e
+                    try:
+                        data_file_manifest_entry = manifest_entries[data_file_old_name]
+                    except KeyError as e:
+                        raise Exception(bundle_fqid, manifest_entry['name'],
+                                        f"Unknown 'file_core.file_name' value '{data_file_old_name}'") from e
+                    updates = {
+                        'file_name': data_file_manifest_entry['new_name'],
+                        'content_type': data_file_manifest_entry['content-type'].partition(';')[0],
+                        'file_size': data_file_manifest_entry['size'],
+                        'file_crc32c': data_file_manifest_entry['crc32c'],
+                        'file_sha1': data_file_manifest_entry['sha1'],
+                        'file_sha256': data_file_manifest_entry['sha256'],
+                    }
+                    if 'checksum' not in file_contents['file_core']:
+                        updates['checksum'] = None
+                    file_contents['file_core'].update(updates)
+                try:
+                    self._upload_file_contents(file_contents, new_name, manifest_entry['content-type'])
+                except Exception as e:
+                    raise Exception(bundle_fqid, manifest_entry['name'], 'Failed to upload file contents') from e
+                if self.args.debug:
+                    benchmarks.append((time.perf_counter() - t1, f"to upload {manifest_entry['name']}"))
+
+        if self.args.debug:
+            total_duration = 0
+            for duration, message in benchmarks:
+                total_duration += duration
+                log.debug(f'{round(duration, 4)} {message}')
+            log.debug(f'{round(total_duration, 4)} Total')
+
+    def _build_blob_key(self, manifest_entry: JSON) -> str:
+        """
+        Return the blob key of the file described by the given manifest_entry.
+        """
+        return '.'.join(manifest_entry[key] for key in ['sha256', 'sha1', 's3_etag', 'crc32c'])
+
+    def _build_new_links_json(self,
+                              old_links_json: JSON,
+                              schema_types: Mapping[str, str]) -> JSON:
+        """
+        Transform the old links.json data into version 2.0.0 syntax.
+        """
+        new_links_json = {
+            'describedBy': 'https://schema.humancellatlas.org/system/2.0.0/links',
+            'schema_type': 'link_bundle',
+            'schema_version': '2.0.0',
+            'links': [],
+        }
+        link: JSON
+        for link in old_links_json['links']:
+            new_link = {
+                'process_id': link['process'],
```

Comment:
This needs to add the new property `link_type`
See ticket https://github.com/HumanCellAtlas/metadata-schema/issues/1285 (PR https://github.com/HumanCellAtlas/metadata-schema/pull/1284)